lib/openssl/ssl.rb

```diff
@@ -117,9 +117,10 @@ class SSLContext
       # If an argument is given, #ssl_version= is called with the value. Note
       # that this form is deprecated. New applications should use #min_version=
       # and #max_version= as necessary.
-      def initialize(version = nil)
+      def initialize(version = nil, fallback_scsv: false)
         self.options |= OpenSSL::SSL::OP_ALL
         self.ssl_version = version if version
+        self.enable_fallback_scsv if fallback_scsv
```

Comment:
I think #initialize should take all or nothing -- it's just confusing to have only fallback_scsv option.

Reply:
`OpenSSL::SSL::SSLContext` is generally used with the short form `SSLContext.new version`, without configuration after that. Having SCSV set on `initialize` allows to keep short form with SCSV protection `SSLContext.new version, fallback_scsv: true` without configuration too.

Reply:
As in the RDoc comment above, use of the form "SSL::SSLContext.new(ssl_method_name)" (and SSL::SSLContext#ssl_version= which is called from it) is not recommended anymore. This is because the version-specific SSL methods has been deprecated in the OpenSSL C API.

In addition to the protocol version which is now preferably set by SSL::SSLContext#{min,max}_version=, clients would usually need to set other options such as verify_mode, too. IMHO SSL::SSLContext has too much options to enumerate in kwargs...

Reply:
Ping on this?